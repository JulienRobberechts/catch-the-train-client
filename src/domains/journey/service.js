import {
  calculateMissionsForJourney,
  formatMissionsSchedules,
} from "./pure/missionSchedule";
import { importMissionsCodes, importMissionsSchedules } from "./dataAccess";

export function getMissionsSchedules() {
  const missionsCodes = importMissionsCodes();
  const missionsSchedules = importMissionsSchedules(missionsCodes);
  return formatMissionsSchedules(missionsSchedules);
}

export default function getMissions(departureStation, arrivalStation) {
  const missionsSchedules = getMissionsSchedules();
  const missions = calculateMissionsForJourney(
    missionsSchedules,
    departureStation,
    arrivalStation
  );
  return missions;
}
