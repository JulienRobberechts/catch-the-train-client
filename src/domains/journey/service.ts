import {
  calculateMissionsForJourney,
  formatMissionsSchedules,
} from "./pure/missionSchedule";
import { getStation } from "./pure/station";
import {
  importMissionsCodes,
  importMissionsSchedules,
  importStations,
} from "./dataAccess";

export function getMissionsSchedules() {
  const missionsCodes = importMissionsCodes();
  const missionsSchedules = importMissionsSchedules(missionsCodes);
  return formatMissionsSchedules(missionsSchedules);
}

export function getMissions(departureStation: string, arrivalStation: string) {
  const missionsSchedules = getMissionsSchedules();
  const missions = calculateMissionsForJourney(
    missionsSchedules,
    departureStation,
    arrivalStation
  );
  return missions;
}

export function getStationBySlug(stationSlug: string | undefined) {
  const allStation = importStations();
  return getStation(stationSlug, allStation);
}

export function getNetworkByKey(networkKey: string) {
  if (!networkKey || networkKey !== "rers") {
    throw Error("invalid networkKey " + networkKey);
  }
  return { key: "rers", name: "RER" };
}

export function getLineByKey(lineKey: string) {
  if (!lineKey || lineKey !== "A") {
    throw Error("invalid lineKey " + lineKey);
  }
  return { key: "A", name: "A" };
}
