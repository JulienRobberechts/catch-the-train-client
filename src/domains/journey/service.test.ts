import each from "jest-each";
import { getStationBySlug } from "./service";

describe("journey services", () => {
  describe("getStationBySlug", () => {
    each([
      ["neuville+universite", "Neuville Universite"],
      ["cergy+le+haut", "Cergy-Le-Haut"],
      ["xxx", undefined],
    ]).test("should get station name for '%s'", (slug, expectedName) => {
      const station = getStationBySlug(slug);
      expect(station?.name).toEqual(expectedName);
    });
  });
});
