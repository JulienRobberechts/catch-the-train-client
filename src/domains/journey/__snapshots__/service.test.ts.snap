// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`journey services getMissionsSchedules should get Missions Schedules 1`] = `
Array [
  Object {
    "mission": "BBTO",
    "stations": Array [
      "torcy",
      "lognes",
      "noisiel",
      "noisy+champs",
      "noisy+le+grand+mont+d'est",
      "bry+sur+marne",
      "neuilly+plaisance",
      "val+de+fontenay",
      "vincennes",
      "nation",
      "gare+de+lyon",
      "chatelet+les+halles",
      "auber",
      "charles+de+gaulle+etoile",
      "grande+arche+la+defense",
    ],
  },
  Object {
    "mission": "BICH",
    "stations": Array [
      "marne+la+vallee+chessy",
      "val+d'europe+serris",
      "bussy+st+georges",
      "torcy",
      "lognes",
      "noisiel",
      "noisy+champs",
      "noisy+le+grand+mont+d'est",
      "bry+sur+marne",
      "neuilly+plaisance",
      "val+de+fontenay",
      "vincennes",
      "nation",
      "gare+de+lyon",
      "chatelet+les+halles",
      "auber",
      "charles+de+gaulle+etoile",
      "grande+arche+la+defense",
    ],
  },
  Object {
    "mission": "TAPI",
    "stations": Array [
      "boissy+saint+leger",
      "sucy+bonneuil",
      "la+varenne+chennevieres",
      "champigny",
      "le+parc+de+saint+maur",
      "saint+maur+creteil",
      "joinville+le+pont",
      "nogent+sur+marne",
      "fontenay+sous+bois",
      "vincennes",
      "nation",
      "gare+de+lyon",
      "chatelet+les+halles",
      "auber",
      "charles+de+gaulle+etoile",
      "grande+arche+la+defense",
      "nanterre+prefecture",
      "houilles",
      "sartrouville",
      "maisons+laffitte",
      "acheres+grand+cormier",
      "poissy",
    ],
  },
  Object {
    "mission": "NANI",
    "stations": Array [
      "cergy+le+haut",
      "cergy+st+christophe",
      "cergy+prefecture",
      "neuville+universite",
      "conflans+fin+d'oise",
      "acheres+ville",
      "maisons+laffitte",
      "sartrouville",
      "houilles",
      "nanterre+prefecture",
      "grande+arche+la+defense",
      "charles+de+gaulle+etoile",
      "auber",
      "chatelet+les+halles",
      "gare+de+lyon",
      "nation",
      "vincennes",
      "fontenay+sous+bois",
      "nogent+sur+marne",
      "joinville+le+pont",
      "saint+maur+creteil",
      "le+parc+de+saint+maur",
      "champigny",
      "la+varenne+chennevieres",
      "sucy+bonneuil",
      "boissy+saint+leger",
    ],
  },
  Object {
    "mission": "NATO",
    "stations": Array [
      "cergy+le+haut",
      "cergy+st+christophe",
      "cergy+prefecture",
      "neuville+universite",
      "conflans+fin+d'oise",
      "acheres+ville",
      "maisons+laffitte",
      "sartrouville",
      "houilles",
      "nanterre+prefecture",
      "grande+arche+la+defense",
      "charles+de+gaulle+etoile",
      "auber",
      "chatelet+les+halles",
      "gare+de+lyon",
      "nation",
      "vincennes",
      "fontenay+sous+bois",
      "nogent+sur+marne",
      "joinville+le+pont",
      "saint+maur+creteil",
      "le+parc+de+saint+maur",
      "champigny",
      "la+varenne+chennevieres",
      "sucy+bonneuil",
      "boissy+saint+leger",
    ],
  },
  Object {
    "mission": "NEGE",
    "stations": Array [
      "st+germain+en+laye",
      "le+vesinet+le+pecq",
      "le+vesinet+centre",
      "chatou+croissy",
      "rueil+malmaison",
      "nanterre+ville",
      "nanterre+universite",
      "nanterre+prefecture",
      "grande+arche+la+defense",
      "charles+de+gaulle+etoile",
      "auber",
      "chatelet+les+halles",
      "gare+de+lyon",
      "nation",
      "vincennes",
      "fontenay+sous+bois",
      "nogent+sur+marne",
      "joinville+le+pont",
      "saint+maur+creteil",
      "le+parc+de+saint+maur",
      "champigny",
      "la+varenne+chennevieres",
      "sucy+bonneuil",
      "boissy+saint+leger",
    ],
  },
  Object {
    "mission": "NELY",
    "stations": Array [
      "st+germain+en+laye",
      "le+vesinet+le+pecq",
      "le+vesinet+centre",
      "chatou+croissy",
      "rueil+malmaison",
      "nanterre+ville",
      "nanterre+universite",
      "nanterre+prefecture",
      "grande+arche+la+defense",
      "charles+de+gaulle+etoile",
      "auber",
      "chatelet+les+halles",
      "gare+de+lyon",
      "nation",
      "vincennes",
      "fontenay+sous+bois",
      "nogent+sur+marne",
      "joinville+le+pont",
      "saint+maur+creteil",
      "le+parc+de+saint+maur",
      "champigny",
      "la+varenne+chennevieres",
      "sucy+bonneuil",
      "boissy+saint+leger",
    ],
  },
  Object {
    "mission": "NEMO",
    "stations": Array [
      "st+germain+en+laye",
      "le+vesinet+le+pecq",
      "le+vesinet+centre",
      "chatou+croissy",
      "rueil+malmaison",
      "nanterre+ville",
      "nanterre+universite",
      "nanterre+prefecture",
      "grande+arche+la+defense",
      "charles+de+gaulle+etoile",
      "auber",
      "chatelet+les+halles",
      "gare+de+lyon",
      "nation",
      "vincennes",
      "fontenay+sous+bois",
      "nogent+sur+marne",
      "joinville+le+pont",
      "saint+maur+creteil",
      "le+parc+de+saint+maur",
      "champigny",
      "la+varenne+chennevieres",
      "sucy+bonneuil",
      "boissy+saint+leger",
    ],
  },
  Object {
    "mission": "NOTE",
    "stations": Array [
      "poissy",
      "acheres+grand+cormier",
      "maisons+laffitte",
      "sartrouville",
      "houilles",
      "nanterre+prefecture",
      "grande+arche+la+defense",
      "charles+de+gaulle+etoile",
      "auber",
      "chatelet+les+halles",
      "gare+de+lyon",
      "nation",
      "vincennes",
      "fontenay+sous+bois",
      "nogent+sur+marne",
      "joinville+le+pont",
      "saint+maur+creteil",
      "le+parc+de+saint+maur",
      "champigny",
      "la+varenne+chennevieres",
      "sucy+bonneuil",
      "boissy+saint+leger",
    ],
  },
  Object {
    "mission": "OKLA",
    "stations": Array [
      "poissy",
      "acheres+grand+cormier",
      "maisons+laffitte",
      "sartrouville",
      "houilles",
      "nanterre+prefecture",
      "grande+arche+la+defense",
      "charles+de+gaulle+etoile",
      "auber",
      "chatelet+les+halles",
      "gare+de+lyon",
      "nation",
      "vincennes",
      "val+de+fontenay",
      "neuilly+plaisance",
      "bry+sur+marne",
      "noisy+le+grand+mont+d'est",
      "noisy+champs",
      "noisiel",
      "lognes",
      "torcy",
    ],
  },
  Object {
    "mission": "OPEN",
    "stations": Array [
      "nanterre+prefecture",
      "grande+arche+la+defense",
      "charles+de+gaulle+etoile",
      "auber",
      "chatelet+les+halles",
      "gare+de+lyon",
      "nation",
      "vincennes",
      "val+de+fontenay",
      "neuilly+plaisance",
      "bry+sur+marne",
      "noisy+le+grand+mont+d'est",
      "noisy+champs",
      "noisiel",
      "lognes",
      "torcy",
    ],
  },
  Object {
    "mission": "QHAR",
    "stations": Array [
      "rueil+malmaison",
      "nanterre+ville",
      "nanterre+universite",
      "nanterre+prefecture",
      "grande+arche+la+defense",
      "charles+de+gaulle+etoile",
      "auber",
      "chatelet+les+halles",
      "gare+de+lyon",
      "nation",
      "vincennes",
      "val+de+fontenay",
      "neuilly+plaisance",
      "bry+sur+marne",
      "noisy+le+grand+mont+d'est",
      "noisy+champs",
      "noisiel",
      "lognes",
      "torcy",
      "bussy+st+georges",
      "val+d'europe+serris",
      "marne+la+vallee+chessy",
    ],
  },
  Object {
    "mission": "QIKI",
    "stations": Array [
      "st+germain+en+laye",
      "le+vesinet+le+pecq",
      "le+vesinet+centre",
      "chatou+croissy",
      "rueil+malmaison",
      "nanterre+ville",
      "nanterre+universite",
      "nanterre+prefecture",
      "grande+arche+la+defense",
      "charles+de+gaulle+etoile",
      "auber",
      "chatelet+les+halles",
      "gare+de+lyon",
      "nation",
      "vincennes",
      "val+de+fontenay",
      "neuilly+plaisance",
      "bry+sur+marne",
      "noisy+le+grand+mont+d'est",
      "noisy+champs",
      "noisiel",
      "lognes",
      "torcy",
      "bussy+st+georges",
      "val+d'europe+serris",
      "marne+la+vallee+chessy",
    ],
  },
  Object {
    "mission": "QIWI",
    "stations": Array [
      "st+germain+en+laye",
      "le+vesinet+le+pecq",
      "le+vesinet+centre",
      "chatou+croissy",
      "rueil+malmaison",
      "nanterre+ville",
      "nanterre+universite",
      "nanterre+prefecture",
      "grande+arche+la+defense",
      "charles+de+gaulle+etoile",
      "auber",
      "chatelet+les+halles",
      "gare+de+lyon",
      "nation",
      "vincennes",
      "val+de+fontenay",
      "neuilly+plaisance",
      "bry+sur+marne",
      "noisy+le+grand+mont+d'est",
      "noisy+champs",
      "noisiel",
      "lognes",
      "torcy",
      "bussy+st+georges",
      "val+d'europe+serris",
      "marne+la+vallee+chessy",
    ],
  },
  Object {
    "mission": "QUDO",
    "stations": Array [
      "poissy",
      "acheres+grand+cormier",
      "maisons+laffitte",
      "sartrouville",
      "houilles",
      "nanterre+prefecture",
      "grande+arche+la+defense",
      "charles+de+gaulle+etoile",
      "auber",
      "chatelet+les+halles",
      "gare+de+lyon",
      "nation",
      "vincennes",
      "val+de+fontenay",
      "neuilly+plaisance",
      "bry+sur+marne",
      "noisy+le+grand+mont+d'est",
      "noisy+champs",
      "noisiel",
      "lognes",
      "torcy",
      "bussy+st+georges",
      "val+d'europe+serris",
      "marne+la+vallee+chessy",
    ],
  },
  Object {
    "mission": "QURE",
    "stations": Array [
      "poissy",
      "acheres+grand+cormier",
      "maisons+laffitte",
      "sartrouville",
      "houilles",
      "nanterre+prefecture",
      "grande+arche+la+defense",
      "charles+de+gaulle+etoile",
      "auber",
      "chatelet+les+halles",
      "gare+de+lyon",
      "nation",
      "vincennes",
      "val+de+fontenay",
      "neuilly+plaisance",
      "bry+sur+marne",
      "noisy+le+grand+mont+d'est",
      "noisy+champs",
      "noisiel",
      "lognes",
      "torcy",
      "bussy+st+georges",
      "val+d'europe+serris",
      "marne+la+vallee+chessy",
    ],
  },
  Object {
    "mission": "QURI",
    "stations": Array [
      "nanterre+prefecture",
      "grande+arche+la+defense",
      "charles+de+gaulle+etoile",
      "auber",
      "chatelet+les+halles",
      "gare+de+lyon",
      "nation",
      "vincennes",
      "val+de+fontenay",
      "neuilly+plaisance",
      "bry+sur+marne",
      "noisy+le+grand+mont+d'est",
      "noisy+champs",
      "noisiel",
      "lognes",
      "torcy",
      "bussy+st+georges",
      "val+d'europe+serris",
      "marne+la+vallee+chessy",
    ],
  },
  Object {
    "mission": "QYAN",
    "stations": Array [
      "cergy+le+haut",
      "cergy+st+christophe",
      "cergy+prefecture",
      "neuville+universite",
      "conflans+fin+d'oise",
      "acheres+ville",
      "maisons+laffitte",
      "sartrouville",
      "houilles",
      "nanterre+prefecture",
      "grande+arche+la+defense",
      "charles+de+gaulle+etoile",
      "auber",
      "chatelet+les+halles",
      "gare+de+lyon",
      "nation",
      "vincennes",
      "val+de+fontenay",
      "neuilly+plaisance",
      "bry+sur+marne",
      "noisy+le+grand+mont+d'est",
      "noisy+champs",
      "noisiel",
      "lognes",
      "torcy",
      "bussy+st+georges",
      "val+d'europe+serris",
      "marne+la+vallee+chessy",
    ],
  },
  Object {
    "mission": "QYLT",
    "stations": Array [
      "cergy+le+haut",
      "cergy+st+christophe",
      "cergy+prefecture",
      "neuville+universite",
      "conflans+fin+d'oise",
      "acheres+ville",
      "maisons+laffitte",
      "sartrouville",
      "houilles",
      "nanterre+prefecture",
      "grande+arche+la+defense",
      "charles+de+gaulle+etoile",
      "auber",
      "chatelet+les+halles",
      "gare+de+lyon",
      "nation",
      "vincennes",
      "val+de+fontenay",
      "neuilly+plaisance",
      "bry+sur+marne",
      "noisy+le+grand+mont+d'est",
      "noisy+champs",
      "noisiel",
      "lognes",
      "torcy",
      "bussy+st+georges",
      "val+d'europe+serris",
      "marne+la+vallee+chessy",
    ],
  },
  Object {
    "mission": "QYRO",
    "stations": Array [
      "nanterre+prefecture",
      "grande+arche+la+defense",
      "charles+de+gaulle+etoile",
      "auber",
      "chatelet+les+halles",
      "gare+de+lyon",
      "nation",
      "vincennes",
      "val+de+fontenay",
      "neuilly+plaisance",
      "bry+sur+marne",
      "noisy+le+grand+mont+d'est",
      "noisy+champs",
      "noisiel",
      "lognes",
      "torcy",
      "bussy+st+georges",
      "val+d'europe+serris",
      "marne+la+vallee+chessy",
    ],
  },
  Object {
    "mission": "TAXE",
    "stations": Array [
      "boissy+saint+leger",
      "sucy+bonneuil",
      "la+varenne+chennevieres",
      "champigny",
      "le+parc+de+saint+maur",
      "saint+maur+creteil",
      "joinville+le+pont",
      "nogent+sur+marne",
      "fontenay+sous+bois",
      "vincennes",
      "nation",
      "gare+de+lyon",
      "chatelet+les+halles",
      "auber",
      "charles+de+gaulle+etoile",
      "grande+arche+la+defense",
      "nanterre+prefecture",
      "houilles",
      "sartrouville",
      "maisons+laffitte",
      "acheres+grand+cormier",
      "poissy",
    ],
  },
  Object {
    "mission": "TEDI",
    "stations": Array [
      "marne+la+vallee+chessy",
      "val+d'europe+serris",
      "bussy+st+georges",
      "torcy",
      "lognes",
      "noisiel",
      "noisy+champs",
      "noisy+le+grand+mont+d'est",
      "bry+sur+marne",
      "neuilly+plaisance",
      "val+de+fontenay",
      "vincennes",
      "nation",
      "gare+de+lyon",
      "chatelet+les+halles",
      "auber",
      "charles+de+gaulle+etoile",
      "grande+arche+la+defense",
      "nanterre+prefecture",
      "houilles",
      "sartrouville",
      "maisons+laffitte",
      "acheres+grand+cormier",
      "poissy",
    ],
  },
  Object {
    "mission": "TERI",
    "stations": Array [
      "marne+la+vallee+chessy",
      "val+d'europe+serris",
      "bussy+st+georges",
      "torcy",
      "lognes",
      "noisiel",
      "noisy+champs",
      "noisy+le+grand+mont+d'est",
      "bry+sur+marne",
      "neuilly+plaisance",
      "val+de+fontenay",
      "vincennes",
      "nation",
      "gare+de+lyon",
      "chatelet+les+halles",
      "auber",
      "charles+de+gaulle+etoile",
      "grande+arche+la+defense",
      "nanterre+prefecture",
      "houilles",
      "sartrouville",
      "maisons+laffitte",
      "acheres+grand+cormier",
      "poissy",
    ],
  },
  Object {
    "mission": "TETE",
    "stations": Array [
      "marne+la+vallee+chessy",
      "val+d'europe+serris",
      "bussy+st+georges",
      "torcy",
      "lognes",
      "noisiel",
      "noisy+champs",
      "noisy+le+grand+mont+d'est",
      "bry+sur+marne",
      "neuilly+plaisance",
      "val+de+fontenay",
      "vincennes",
      "nation",
      "gare+de+lyon",
      "chatelet+les+halles",
      "auber",
      "charles+de+gaulle+etoile",
      "grande+arche+la+defense",
      "nanterre+prefecture",
      "houilles",
      "sartrouville",
      "maisons+laffitte",
      "acheres+grand+cormier",
      "poissy",
    ],
  },
  Object {
    "mission": "TROC",
    "stations": Array [
      "torcy",
      "lognes",
      "noisiel",
      "noisy+champs",
      "noisy+le+grand+mont+d'est",
      "bry+sur+marne",
      "neuilly+plaisance",
      "val+de+fontenay",
      "vincennes",
      "nation",
      "gare+de+lyon",
      "chatelet+les+halles",
      "auber",
      "charles+de+gaulle+etoile",
      "grande+arche+la+defense",
      "nanterre+prefecture",
      "houilles",
      "sartrouville",
      "maisons+laffitte",
      "acheres+grand+cormier",
      "poissy",
    ],
  },
  Object {
    "mission": "UPAC",
    "stations": Array [
      "marne+la+vallee+chessy",
      "val+d'europe+serris",
      "bussy+st+georges",
      "torcy",
      "lognes",
      "noisiel",
      "noisy+champs",
      "noisy+le+grand+mont+d'est",
      "bry+sur+marne",
      "neuilly+plaisance",
      "val+de+fontenay",
      "vincennes",
      "nation",
      "gare+de+lyon",
      "chatelet+les+halles",
      "auber",
      "charles+de+gaulle+etoile",
      "grande+arche+la+defense",
      "nanterre+prefecture",
      "houilles",
      "sartrouville",
      "maisons+laffitte",
      "acheres+ville",
      "conflans+fin+d'oise",
      "neuville+universite",
      "cergy+prefecture",
      "cergy+st+christophe",
      "cergy+le+haut",
    ],
  },
  Object {
    "mission": "UPUS",
    "stations": Array [
      "marne+la+vallee+chessy",
      "val+d'europe+serris",
      "bussy+st+georges",
      "torcy",
      "lognes",
      "noisiel",
      "noisy+champs",
      "noisy+le+grand+mont+d'est",
      "bry+sur+marne",
      "neuilly+plaisance",
      "val+de+fontenay",
      "vincennes",
      "nation",
      "gare+de+lyon",
      "chatelet+les+halles",
      "auber",
      "charles+de+gaulle+etoile",
      "grande+arche+la+defense",
      "nanterre+prefecture",
      "houilles",
      "sartrouville",
      "maisons+laffitte",
      "acheres+ville",
      "conflans+fin+d'oise",
      "neuville+universite",
      "cergy+prefecture",
      "cergy+st+christophe",
      "cergy+le+haut",
    ],
  },
  Object {
    "mission": "UZAR",
    "stations": Array [
      "boissy+saint+leger",
      "sucy+bonneuil",
      "la+varenne+chennevieres",
      "champigny",
      "le+parc+de+saint+maur",
      "saint+maur+creteil",
      "joinville+le+pont",
      "nogent+sur+marne",
      "fontenay+sous+bois",
      "vincennes",
      "nation",
      "gare+de+lyon",
      "chatelet+les+halles",
      "auber",
      "charles+de+gaulle+etoile",
      "grande+arche+la+defense",
      "nanterre+prefecture",
      "houilles",
      "sartrouville",
      "maisons+laffitte",
      "acheres+ville",
      "conflans+fin+d'oise",
      "neuville+universite",
      "cergy+prefecture",
      "cergy+st+christophe",
      "cergy+le+haut",
    ],
  },
  Object {
    "mission": "UZEL",
    "stations": Array [
      "boissy+saint+leger",
      "sucy+bonneuil",
      "la+varenne+chennevieres",
      "champigny",
      "le+parc+de+saint+maur",
      "saint+maur+creteil",
      "joinville+le+pont",
      "nogent+sur+marne",
      "fontenay+sous+bois",
      "vincennes",
      "nation",
      "gare+de+lyon",
      "chatelet+les+halles",
      "auber",
      "charles+de+gaulle+etoile",
      "grande+arche+la+defense",
      "nanterre+prefecture",
      "houilles",
      "sartrouville",
      "maisons+laffitte",
      "acheres+ville",
      "conflans+fin+d'oise",
      "neuville+universite",
      "cergy+prefecture",
      "cergy+st+christophe",
      "cergy+le+haut",
    ],
  },
  Object {
    "mission": "YRZZ",
    "stations": Array [
      "la+varenne+chennevieres",
      "champigny",
      "le+parc+de+saint+maur",
      "saint+maur+creteil",
      "joinville+le+pont",
      "nogent+sur+marne",
      "fontenay+sous+bois",
      "vincennes",
      "nation",
      "gare+de+lyon",
      "chatelet+les+halles",
      "auber",
      "charles+de+gaulle+etoile",
      "grande+arche+la+defense",
      "nanterre+prefecture",
      "nanterre+universite",
      "nanterre+ville",
      "rueil+malmaison",
    ],
  },
  Object {
    "mission": "ZEBU",
    "stations": Array [
      "boissy+saint+leger",
      "sucy+bonneuil",
      "la+varenne+chennevieres",
      "champigny",
      "le+parc+de+saint+maur",
      "saint+maur+creteil",
      "joinville+le+pont",
      "nogent+sur+marne",
      "fontenay+sous+bois",
      "vincennes",
      "nation",
      "gare+de+lyon",
      "chatelet+les+halles",
      "auber",
      "charles+de+gaulle+etoile",
      "grande+arche+la+defense",
      "nanterre+prefecture",
      "nanterre+universite",
      "nanterre+ville",
      "rueil+malmaison",
      "chatou+croissy",
      "le+vesinet+centre",
      "le+vesinet+le+pecq",
      "st+germain+en+laye",
    ],
  },
  Object {
    "mission": "ZEMA",
    "stations": Array [
      "boissy+saint+leger",
      "sucy+bonneuil",
      "la+varenne+chennevieres",
      "champigny",
      "le+parc+de+saint+maur",
      "saint+maur+creteil",
      "joinville+le+pont",
      "nogent+sur+marne",
      "fontenay+sous+bois",
      "vincennes",
      "nation",
      "gare+de+lyon",
      "chatelet+les+halles",
      "auber",
      "charles+de+gaulle+etoile",
      "grande+arche+la+defense",
      "nanterre+prefecture",
      "nanterre+universite",
      "nanterre+ville",
      "rueil+malmaison",
      "chatou+croissy",
      "le+vesinet+centre",
      "le+vesinet+le+pecq",
      "st+germain+en+laye",
    ],
  },
  Object {
    "mission": "ZEUS",
    "stations": Array [
      "boissy+saint+leger",
      "sucy+bonneuil",
      "la+varenne+chennevieres",
      "champigny",
      "le+parc+de+saint+maur",
      "saint+maur+creteil",
      "joinville+le+pont",
      "nogent+sur+marne",
      "fontenay+sous+bois",
      "vincennes",
      "nation",
      "gare+de+lyon",
      "chatelet+les+halles",
      "auber",
      "charles+de+gaulle+etoile",
      "grande+arche+la+defense",
      "nanterre+prefecture",
      "nanterre+universite",
      "nanterre+ville",
      "rueil+malmaison",
      "chatou+croissy",
      "le+vesinet+centre",
      "le+vesinet+le+pecq",
      "st+germain+en+laye",
    ],
  },
  Object {
    "mission": "ZULI",
    "stations": Array [
      "marne+la+vallee+chessy",
      "val+d'europe+serris",
      "bussy+st+georges",
      "torcy",
      "lognes",
      "noisiel",
      "noisy+champs",
      "noisy+le+grand+mont+d'est",
      "bry+sur+marne",
      "neuilly+plaisance",
      "val+de+fontenay",
      "vincennes",
      "nation",
      "gare+de+lyon",
      "chatelet+les+halles",
      "auber",
      "charles+de+gaulle+etoile",
      "grande+arche+la+defense",
      "nanterre+prefecture",
      "nanterre+universite",
      "nanterre+ville",
      "rueil+malmaison",
      "chatou+croissy",
      "le+vesinet+centre",
      "le+vesinet+le+pecq",
      "st+germain+en+laye",
    ],
  },
  Object {
    "mission": "ZUPE",
    "stations": Array [
      "marne+la+vallee+chessy",
      "val+d'europe+serris",
      "bussy+st+georges",
      "torcy",
      "lognes",
      "noisiel",
      "noisy+champs",
      "noisy+le+grand+mont+d'est",
      "bry+sur+marne",
      "neuilly+plaisance",
      "val+de+fontenay",
      "vincennes",
      "nation",
      "gare+de+lyon",
      "chatelet+les+halles",
      "auber",
      "charles+de+gaulle+etoile",
      "grande+arche+la+defense",
      "nanterre+prefecture",
      "nanterre+universite",
      "nanterre+ville",
      "rueil+malmaison",
      "chatou+croissy",
      "le+vesinet+centre",
      "le+vesinet+le+pecq",
      "st+germain+en+laye",
    ],
  },
  Object {
    "mission": "ZITA",
    "stations": Array [
      "torcy",
      "noisiel",
      "noisy+champs",
      "noisy+le+grand+mont+d'est",
      "bry+sur+marne",
      "val+de+fontenay",
      "vincennes",
      "nation",
      "gare+de+lyon",
      "chatelet+les+halles",
      "auber",
      "charles+de+gaulle+etoile",
      "grande+arche+la+defense",
      "nanterre+prefecture",
      "nanterre+universite",
      "nanterre+ville",
      "rueil+malmaison",
      "chatou+croissy",
      "le+vesinet+centre",
      "le+vesinet+le+pecq",
      "st+germain+en+laye",
    ],
  },
  Object {
    "mission": "ZBRA",
    "stations": Array [
      "marne+la+vallee+chessy",
      "val+d'europe+serris",
      "torcy",
      "lognes",
      "noisy+champs",
      "noisy+le+grand+mont+d'est",
      "bry+sur+marne",
      "neuilly+plaisance",
      "val+de+fontenay",
      "vincennes",
      "nation",
      "gare+de+lyon",
      "chatelet+les+halles",
      "auber",
      "charles+de+gaulle+etoile",
      "grande+arche+la+defense",
      "nanterre+prefecture",
      "nanterre+universite",
      "nanterre+ville",
      "rueil+malmaison",
      "chatou+croissy",
      "le+vesinet+centre",
      "le+vesinet+le+pecq",
      "st+germain+en+laye",
    ],
  },
  Object {
    "mission": "ZCAR",
    "stations": Array [
      "marne+la+vallee+chessy",
      "val+d'europe+serris",
      "bussy+st+georges",
      "torcy",
      "lognes",
      "noisiel",
      "noisy+champs",
      "noisy+le+grand+mont+d'est",
      "neuilly+plaisance",
      "val+de+fontenay",
      "vincennes",
      "nation",
      "gare+de+lyon",
      "chatelet+les+halles",
      "auber",
      "charles+de+gaulle+etoile",
      "grande+arche+la+defense",
      "nanterre+prefecture",
      "nanterre+universite",
      "nanterre+ville",
      "rueil+malmaison",
      "chatou+croissy",
      "le+vesinet+centre",
      "le+vesinet+le+pecq",
      "st+germain+en+laye",
    ],
  },
  Object {
    "mission": "ONIX",
    "stations": Array [
      "st+germain+en+laye",
      "le+vesinet+le+pecq",
      "le+vesinet+centre",
      "chatou+croissy",
      "rueil+malmaison",
      "nanterre+ville",
      "nanterre+universite",
      "grande+arche+la+defense",
      "charles+de+gaulle+etoile",
      "auber",
      "chatelet+les+halles",
      "gare+de+lyon",
      "nation",
      "vincennes",
      "val+de+fontenay",
      "neuilly+plaisance",
      "bry+sur+marne",
      "noisy+le+grand+mont+d'est",
      "noisy+champs",
      "noisiel",
      "lognes",
      "torcy",
    ],
  },
  Object {
    "mission": "QCOQ",
    "stations": Array [
      "le+vesinet+le+pecq",
      "le+vesinet+centre",
      "chatou+croissy",
      "rueil+malmaison",
      "nanterre+ville",
      "nanterre+universite",
      "nanterre+prefecture",
      "grande+arche+la+defense",
      "charles+de+gaulle+etoile",
      "auber",
      "chatelet+les+halles",
      "gare+de+lyon",
      "nation",
      "vincennes",
      "val+de+fontenay",
      "neuilly+plaisance",
      "bry+sur+marne",
      "noisy+le+grand+mont+d'est",
      "noisy+champs",
      "noisiel",
      "torcy",
      "bussy+st+georges",
      "val+d'europe+serris",
      "marne+la+vallee+chessy",
    ],
  },
  Object {
    "mission": "QNOR",
    "stations": Array [
      "st+germain+en+laye",
      "le+vesinet+le+pecq",
      "le+vesinet+centre",
      "chatou+croissy",
      "rueil+malmaison",
      "nanterre+ville",
      "nanterre+universite",
      "nanterre+prefecture",
      "grande+arche+la+defense",
      "charles+de+gaulle+etoile",
      "auber",
      "chatelet+les+halles",
      "gare+de+lyon",
      "nation",
      "vincennes",
      "val+de+fontenay",
      "neuilly+plaisance",
      "noisy+le+grand+mont+d'est",
      "noisy+champs",
      "lognes",
      "torcy",
      "bussy+st+georges",
      "val+d'europe+serris",
      "marne+la+vallee+chessy",
    ],
  },
  Object {
    "mission": "QBIK",
    "stations": Array [
      "st+germain+en+laye",
      "le+vesinet+le+pecq",
      "le+vesinet+centre",
      "chatou+croissy",
      "rueil+malmaison",
      "nanterre+ville",
      "nanterre+universite",
      "nanterre+prefecture",
      "grande+arche+la+defense",
      "charles+de+gaulle+etoile",
      "auber",
      "chatelet+les+halles",
      "gare+de+lyon",
      "nation",
      "vincennes",
      "val+de+fontenay",
      "neuilly+plaisance",
      "bry+sur+marne",
      "noisy+le+grand+mont+d'est",
      "noisy+champs",
      "noisiel",
      "torcy",
      "bussy+st+georges",
      "val+d'europe+serris",
      "marne+la+vallee+chessy",
    ],
  },
]
`;
